src/ir/possible-contents.h

```diff
@@ -213,23 +214,22 @@ class PossibleContents {
   }
 
   size_t hash() const {
-    // Encode this using three bits for the variant type, then the rest of the
-    // contents.
-    if (isNone()) {
-      return 0;
+    // First hash the index of the variant, then add the internals for each.
+    size_t ret = std::hash<size_t>()(value.index());
+    if (isNone() || isMany()) {
+      // Nothing to add.
     } else if (isLiteral()) {
-      return size_t(1) | (std::hash<Literal>()(getLiteral()) << 3);
+      hash_combine(ret, std::hash<Literal>()(getLiteral()));
```

Comment:
I think it would be slightly simpler to use `rehash(ret, getLiteral())` and similar elsewhere. This could also be implemented via `std::visit([&](auto val) { rehash(ret, val); }, value)`, but that would require hash implementations for none and many as well.

Reply:
Oh nice, I'd forgotten about `rehash` I guess... done.